app/scripts/modules/amazon/src/subnet/subnetSelectField.component.ts

```diff
@@ -53,7 +53,7 @@ class SubnetSelectFieldController implements IController {
     this.activeSubnets = subnets.filter(s => !s.deprecated);
     this.deprecatedSubnets = subnets.filter(s => s.deprecated);
     if (subnets.length) {
-      if (!this.component[this.field] && !this.readOnly) {
+      if (this.component[this.field] === null && !this.readOnly) {
```

Comment:
What was this change fixing?

Reply:
The subnetType is initially null and is changed to "" (empty string) for EC2 classic. Without this, `this.change()` is called which in turns recomputes the subnets (loop).

I should say that I don't think this directly fixes 3242 as I wrote in the summary. It's already more or less fixed but I'm preparing fixes for release branches as well where that issue still exist.